### ISSUE TITLE
Teleporting and penetration issues

### ISSUE THREAD

**Issue by vvs-** [opentomb/OpenTomb#458]
In _The Hive_ position 40000,29000,-3360,90,0,0 there is a bug which have similarity with several other bugs at once. If you try to use action+up Lara climbs up through wall. But if you use action+jump Lara is teleported and grabs on the floor from below the next sector.

**Comment by vvs-** [comment]
Also, climbing from water room 39 teleports to adjacent room 10 instead of 40.

**Comment by vvs-** [comment]
Ok. Looks like the old case of badly intertwined overlapping rooms.

**Comment by TeslaRus** [comment]
I have idea about extra conditions checking...

**Comment by vvs-** [comment]
Room 29 is fixed, but rooms 39/40 still teleport to room 10.

**Comment by TeslaRus** [comment]
+there was another bug: in same place (The Hive position 40000,29000,-3360,90,0,0) jump up + ctrl prevents to underground hanging: fixed by frame > 2 in Lara's state control

**Comment by vvs-** [comment]
Try `setEntityPos(155,87095,60867,1792,90,0,0)` in _Opera House_. I've got that old annoying animation and even with wall penetration.

**Comment by TeslaRus** [comment]
60 fps locked, only jump up higher edge, no teleport and over the level... I will try co configure offsets in state control

**Comment by vvs-** [comment]
Yes, it depends on frame rate. If there is no penetration just use `setEntytPos` several times, because there is some non-deterministic factor which sometimes changes position on its first run.
